common/src/main/java/org/jboss/gm/common/utils/PluginUtils.java

```diff
@@ -0,0 +1,111 @@
+package org.jboss.gm.common.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.commonjava.maven.ext.common.ManipulationException;
+import org.jboss.gm.common.Configuration;
+import org.slf4j.Logger;
+
+public class PluginUtils {
+    private static final Map<String, String> SUPPORTED_PLUGINS = new HashMap<>();
+    private static final List<String> BUILD_FILES = Arrays.asList("build.gradle", "settings.gradle", "build.gradle.kts",
+            "settings.gradle.kts");
+
+    static {
+        SUPPORTED_PLUGINS.put("com.github.ben-manes.versions", "dependencyUpdates");
+        SUPPORTED_PLUGINS.put("com.github.burrunan.s3-build-cache", "buildCache");
+        SUPPORTED_PLUGINS.put("gradle-enterprise", "gradleEnterprise");
+    }
+
+    /**
+     * Removes plugins from a target build file.
+     *
+     * Will automatically examine build.gradle, settings.gradle, build.gradle.kts, settings.gradle.kts
+     *
+     * @param logger the current logger in use
+     * @param configuration the current configuration
+     * @param t the build target directory
+     */
+    public static void pluginRemoval(Logger logger, Configuration configuration, File t)
+            throws ManipulationException {
+        final String[] pluginRemoval = configuration.pluginRemoval();
+
+        if (pluginRemoval != null) {
+            for (String plugin : pluginRemoval) {
+                if (!SUPPORTED_PLUGINS.containsKey(plugin)) {
+                    throw new ManipulationException("No support for removing plugin {}", plugin);
+                }
+                for (String b : BUILD_FILES) {
+                    File buildFile = new File(t, b);
+
+                    if (!buildFile.exists()) {
+                        logger.debug("No {} found in {}", b, t);
+                        continue;
+                    }
+
+                    try {
+                        org.apache.commons.io.FileUtils.readLines(buildFile, Charset.defaultCharset());
+                        String target = org.apache.commons.io.FileUtils.readFileToString(buildFile, Charset.defaultCharset());
+                        String configTask = SUPPORTED_PLUGINS.get(plugin);
+
+                        logger.debug("Looking to remove {} with configuration block of {}", plugin, configTask);
+                        if (target.contains(plugin)) {
+
+                            List<String> lines = new ArrayList<>(Arrays.asList(target.split("\\R")));
```

Comment:
In this different from using `org.apache.commons.io.FileUtils.readLines` directly instead of using `org.apache.commons.io.FileUtils.readFileToString`?

Reply:
Yes, I needed it both as individual lines and as a entire string. I'll clean it up a bit